test/extended/baremetal/high_availability.go

```diff
@@ -0,0 +1,117 @@
+package baremetal
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	interval           = 10 * time.Second
+	waitTimeout        = 5 * time.Minute
+	baremetalNamespace = "openshift-machine-api"
+)
+
+var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platform should ensure", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("baremetal")
+
+	g.It("cluster baremetal operator returns back healthy after it is deleted", func() {
+		skipIfNotBaremetal(oc)
+
+		c, err := e2e.LoadClientset()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("delete cluster baremetal operator")
+		err = c.AppsV1().Deployments(baremetalNamespace).Delete(context.Background(), "cluster-baremetal-operator", v1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("wait until cluster baremetal operator returns back healthy")
+		err = wait.Poll(interval, waitTimeout, func() (bool, error) {
```

Comment:
I think that In case of a slow environment this single wait loop could miss the delete.
I'd suggest to add first a `wait.PollImmediate` (and checking until `apierrors.IsNotFound`),
and then another `wait.PollImmediate` until the new deployment is found

Reply:
Good point @andfasano